loadtest/control/simulcontroller/actions.go

```diff
@@ -33,3 +39,108 @@ func (c *SimulController) reload(full bool) control.UserActionResponse {
 
 	return control.Reload(c.user)
 }
+
+func (c *SimulController) joinTeam(u user.User) control.UserActionResponse {
+	userStore := u.Store()
+	userId := userStore.Id()
+	teamIds, err := u.GetAllTeams(0, 100)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	for _, teamId := range teamIds {
+		tm, err := userStore.TeamMember(teamId, userId)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		if tm.UserId == "" {
+			if err := u.AddTeamMember(teamId, userId); err != nil {
+				return control.UserActionResponse{Err: control.NewUserError(err)}
+			}
+			c.status <- c.newInfoStatus(fmt.Sprintf("joined team %s", teamId))
+			break
+		}
+	}
+	return c.switchTeam(u)
+}
+
+func (c *SimulController) switchTeam(u user.User) control.UserActionResponse {
+	// TODO: change this to make sure we don't pick the same team we are
+	// currently in.
+	team, err := u.Store().RandomTeamJoined()
```

Comment:
We can now update this to `RandomTeam(store.SelectMemberOf)` Also, we can add a new selection criteria, `store.CurrentlyIn` :)

Reply:
We have too :D